src/utils.js

```diff
@@ -381,3 +381,23 @@ export const focusableElements = [
   'video[controls]',
   '[contenteditable]:not([contenteditable="false"])',
 ];
+
+export function addDefaultProps(defaultProps, props) {
+  if (!defaultProps || !props) return props;
+
+  let result = { ...props };
+
+  Object.keys(defaultProps).forEach((key) => {
+    if (result[key] === undefined) {
+      result[key] = defaultProps[key];
+    }
+    if (
+      Object.keys(defaultProps[key] || {}).length > 0 &&
+      typeof defaultProps[key] === 'object'
```

Comment:
This is not handling the array and the `null` case correctly. We do not need to merge if it is an array so the conditions should be somwhere around the lines of `!Array.isArray(defaultProps[key])` and should also have a check for `null` case `defaultProps[key] !== null` because `typeof null === 'object'` is true